lesson10/main.js

```diff
@@ -0,0 +1,52 @@
+const ul = document.getElementById("js-ul");
+const loading = document.getElementById("js-loading")
+
+const addLoading = () =>{
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () =>{
+  document.getElementById('loading-img').remove();
+};
+
+const createLists = (features) =>{
+  const fragment = new DocumentFragment();
+
+  for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt;
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    };
+
+    ul.appendChild(fragment);
+};
+
+
+const receiveData = new Promise((resolve) =>{
+  addLoading();
```

Comment:
IMO
This may be a personal feeling, but I felt it was better not to include loading in the Prosmise object. I felt it was better to simply only let the value resolve here.
Why not move it to asyncFeatures()?

```
const asyncFeatures = async () => {
  addLoading();
  try {
    const result = await receiveData;
    createLists(result);
  } catch (error) {
    console.error(error);
  } finally {
    removeLoading();
  }
};
```

Reply:
Thank you Chihiro-san,
removed addLoading out of promise method as per your suggestion